L1Trigger/TrackerTFP/interface/DataFormats.h

```diff
@@ -0,0 +1,653 @@
+#ifndef L1Trigger_TrackerTFP_DataFormats_h
+#define L1Trigger_TrackerTFP_DataFormats_h
+
+/*----------------------------------------------------------------------
+Classes to calculate and provide dataformats used by Track Trigger emulator
+enabling automated conversions from frames to stubs/tracks and vice versa
+----------------------------------------------------------------------*/
+
+#include "FWCore/Framework/interface/data_default_record_trait.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "L1Trigger/TrackerTFP/interface/DataFormatsRcd.h"
+#include "L1Trigger/TrackerDTC/interface/Setup.h"
+#include "DataFormats/L1TrackTrigger/interface/TTBV.h"
+
+#include <vector>
+#include <cmath>
+#include <initializer_list>
+#include <tuple>
+#include <iostream>
+
+namespace trackerTFP {
+
+  // track trigger processes
+  enum class Process { begin, fe = begin, dtc, pp, gp, ht, mht, sf, kfin, kf, dr, end, x };
+  // track trigger variables
+  enum class Variable { begin, r = begin, phi, z, layer, sectorsPhi, sectorEta, sectorPhi, phiT, qOverPt, zT, cot, trackId, match, hitPattern, layerMap, phi0, z0, end, x };
+  // track trigger process order
+  constexpr std::initializer_list<Process> Processes = {Process::fe, Process::dtc, Process::pp, Process::gp, Process::ht, Process::mht, Process::sf, Process::kfin, Process::kf, Process::dr};
+  // conversion: Process to int
+  inline constexpr int operator+(Process p) { return static_cast<int>(p); }
+  // conversion: Variable to int
+  inline constexpr int operator+(Variable v) { return static_cast<int>(v); }
```

Comment:
Naming this function "operator+" is misleading, since it does no addition. Please using another name, such as "Int" instead.

Reply:
Using the unary operator is not uncommon to convert enums into ints. I would like to use an operator instead of a function to get more compact code. From all operators I can overload that one seem to be most suitable.

Reply:
When reading code, people will often not take the time to investigate what every individual function call does, but will simply guess what functions do, based on their name. So using function names that mislead the user must be avoided. If this "operator+" is only used in the few cases I see inside DataFormats, then please use Int() instead. Making the code comprehensible is more important than making it compact. If it is used in numerous cases, and using Int() would significantly increase the length, then see if you can find a less misleading operator function.